cpp_utils/src/cpp/time/time_utils.cpp

```diff
@@ -36,17 +36,25 @@ namespace utils {
 
 Timestamp now() noexcept
 {
-    return std::chrono::system_clock::now();
+    return Timeclock::now();
 }
 
 Timestamp the_end_of_time() noexcept
 {
-    return std::chrono::time_point<std::chrono::system_clock>::max();
+    #if _EPROSIMA_WINDOWS_PLATFORM // In Windows std::gmtime does not support negative values nor values greater than 2^32
```

Comment:
I don't think we should be constraining precision due to limitations in a method we are only using for time to calendar conversions. I believe we should address the limitation in that method only, hoping the fix the problem in the future or we might as well change the implementation to not use that problematic method.